### PR TITLE
feat: scoped registries

### PR DESCRIPTION
**Pull request by nvandessel**
#### Changes

- Add documentation relevant to the use of `scopedRegistryUrl` and `registryScopes`
- Add documentation for private scoped registries and using the `UPM_REGISTRY_TOKEN` environment variable

#### Checklist

- [x] Read the contribution [guide](../blob/main/CONTRIBUTING.md) and accept the
      [code](../blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

File: docs/03-github/03-test-runner.mdx [hunk 1]
```diff
@@ -94,6 +94,38 @@ the [`packageMode`](#packagemode) configuration option to `true`.
...
+
+```yaml
+- uses: game-ci/unity-test-runner@v4
+  env:
+    UNITY_EMAIL: ${{ secrets.UNITY_EMAIL }}
+    UNITY_PASSWORD: ${{ secrets.UNITY_PASSWORD }}
+    UNITY_SERIAL: ${{ secrets.UNITY_SERIAL }}
+    UPM_REGISTRY_TOKEN: ${{ secrets.UPM_REGISTRY_TOKEN }}
+  with:
+    pscopedRegistryUrl: https://example.com/registry
```
Comment (GabLeRoux) [inline docs/03-github/03-test-runner.mdx:125]: I think there's a typo here

```suggestion
    scopedRegistryUrl: https://example.com/registry
```

**Review by GabLeRoux — APPROVED** [review]
should be merged only when related PR is merged 👍 Nice work.

**Comment by nvandessel** [comment]
Thank you so much! Appreciate the keen eye finding that typo! 😬

**Review by webbertakken — APPROVED** [review]

**Comment by GabLeRoux** [comment]
Let's merge this. Only missing thing to do is to fix code formatting with `yarn lint`. 👍

**Comment by GabLeRoux** [comment]
I wasn't able to update your branch, but I was able to create a new branch in this repository here: https://github.com/game-ci/documentation/tree/feature/scoped-registries

**Pull request merged**